### DISCUSSION TITLE
what is the 7.4.3 upgrade path to 7.7.1?

### DISCUSSION THREAD

**Discussion by jmccann** [concourse/concourse#8298] [Help & Support] [answered]
If we are running 7.4.3 and looking to upgrade to a newer version, say 7.7.1, what is the upgrade path?

In 7.4.3 it says

> If you are on 7.4.1 or 7.4.2, you will need to downgrade back to 7.4.0 using backups you have or the script 7.4.x-down-migration.sh included in the assets of this release. Only after downgrading can you safely upgrade directly to 7.4.3 or any later version of Concourse

So we upgraded to 7.4.3 (from 7.1 or something) to be sure we could upgrade to later versions fine.  

But now I see 7.4.4 which says

> This PR fixes a typo in a down migration that affects Concourse version 7.4.1, 7.4.2 and 7.4.3. Therefore, you will be unable to downgrade from any of those listed versions. If you are currently on 7.4.1, 7.4.2 or 7.4.3, you will need to downgrade back to 7.4.0 before you can upgrade to this version of 7.4.4. The instructions on how to downgrade to 7.4.0 is documented in the release notes of the 7.4.x version you are on.

So I'm not sure if that is saying I can upgrade to 7.7.1 fine from 7.4.3 ... its only if I want to get to 7.4.4 I would first have to downgrade to 7.4.0?

Or do I first need to downgrade to 7.4.0 before going to 7.7.1?

Or do I need to downgrade to 7.4.0, upgrade to 7.4.4 then I can go to 7.7.1?

**Comment by xtremerui** [comment, marked as answer]
> So I'm not sure if that is saying I can upgrade to 7.7.1 fine from 7.4.3 ... its only if I want to get to 7.4.4 I would first have to downgrade to 7.4.0?

This is correct one. In any version you can upgrade to 7.7.1 without problem.

**Reply by jmccann** [reply]
Thanks so much for the reply!

**Reply by wanderanimrod** [reply]
@xtremerui , this doesn't seem to be true. We upgraded from 7.4.0 to 7.7.1 and we are getting that migration error:
```
{"build":"check","error":"failed to find pipeline: pq: column p.paused_by does not exist" ...
```
We had never upgraded to any 7.4.x version. What's the path for migrating from 7.4.0 to 7.7.1?

**Reply by xtremerui** [reply]
https://github.com/concourse/concourse/discussions/7746#discussioncomment-1753894 according to the comment and also checking the migrations file (https://github.com/concourse/concourse/tree/v7.4.0/atc/db/migration/migrations
https://github.com/concourse/concourse/tree/v7.7.1/atc/db/migration/migrations) I dont see a probelm going from 7.4.0 to 7.7.1

You could verify your DB version in `migrations_history` table. For v7.4.0, the lastest `version` should be `1625844436` and its `status` should be `passed`.

**Reply by xtremerui** [reply]
Another thing to verify is to check whether `paused_at` and `paused_by` exists in `jobs` table.

**Reply by wanderanimrod** [reply]
Sorry about the confusion @xtremerui. I looked at the `migration_history` table and found that the latest migration version was `1627322257`, which was part of `7.4.1` and `7.4.2` but not 7.4.3. We were running concourse version `7.4.0`, so I think there was a botched migration to `7.4.2` or `7.4.1` on the cluster in the past which wasn't properly rolled back. That's why the `migration_history` table was in that state. 

To fix this, I ran [the `7.4.x-down-migration.sh` script](https://github.com/concourse/concourse/releases/download/v7.4.1/7.4.x-down-migration.sh) from version `7.4.1` against the database. This reset it to the correct 7.4.0 migration state (with `1625844436` being the latest migration). Once I did that, I could migrate just fine.

Thanks for looking into this.
